base/asyncevent.jl

```diff
@@ -115,6 +124,7 @@ function _trywait(t::Union{Timer, AsyncCondition})
         # full barrier now for AsyncCondition
         t isa Timer || Core.Intrinsics.atomic_fence(:acquire_release)
     else
+        t.isopen || return false
```

Comment:
Since this is outside a lock, don't we need to acquire load `t.isopen` for `AsyncCondition` so that `close(async)` happens-before `wait(async)` returns?

Reply:
I think it's conceivable that someone writes

```julia
async = AsyncCondition()

t1 = @spawn begin  # code in external library
    f1()
    ccall(:uv_async_send, Cvoid, (Ptr{Cvoid},), async)
end

t2 = @spawn begin
    try
        wait(async)
    catch err
        err isa EOFError || rethrow()
    end
    close(async)
    f2()
end

t3 = @spawn begin
    try
        wait(async)
    catch err
        err isa EOFError || rethrow()
    end
    close(async)
    f3()
end
```

and expect `f1` to happen-before `f2` and `f3`. There are better ways but it does not look super crazy to me.

But I'm also OK with documenting that `wait` does not establish any ordering if it throws.

Reply:
Establishing happens-before seems like a good idea. We don't do that currently, so it is not relevant to this PR however.